src/sentry/scim/endpoints/users.py

```diff
@@ -0,0 +1,154 @@
+
+from .constants import SCIM_400_INVALID_FILTER, SCIM_409_USER_EXISTS
+from .utils import SCIMEndpoint, parse_filter_conditions
+
+ERR_ONLY_OWNER = "You cannot remove the only remaining owner of the organization."
+from rest_framework.exceptions import PermissionDenied
+
+from sentry.api.exceptions import ConflictError
+
+
+class OrganizationSCIMUserDetails(SCIMEndpoint, OrganizationMemberEndpoint):
+    def _delete_member(self, request, organization, member):
+        audit_data = member.get_audit_log_data()
+        if OrganizationMemberDetailsEndpoint._is_only_owner(member):
+            raise PermissionDenied(detail=ERR_ONLY_OWNER)
+        with transaction.atomic():
+            AuthIdentity.objects.filter(
+                user=member.user, auth_provider__organization=organization
+            ).delete()
+            member.delete()
+            self.create_audit_entry(
+                request=request,
+                organization=organization,
+                target_object=member.id,
+                target_user=member.user,
+                event=AuditLogEntryEvent.MEMBER_REMOVE,
+                data=audit_data,
+            )
+
+    def get(self, request, organization, member):
+        context = serialize(member, serializer=OrganizationMemberSCIMSerializer())
+        return Response(context)
+
+    def patch(self, request, organization, member):
+        for operation in request.data.get("Operations", []):
+            # we only support setting active to False which deletes the orgmember
+            if operation["value"]["active"] is False:
+                self._delete_member(request, organization, member)
+                return Response(status=204)
+        context = serialize(member, serializer=OrganizationMemberSCIMSerializer())
+        return Response(context)
+
+    def delete(self, request, organization, member):
+        self._delete_member(request, organization, member)
+        return Response(status=204)
+
+
+class OrganizationSCIMUserIndex(SCIMEndpoint):
+    def get(self, request, organization):
+        # note that SCIM doesn't care about changing results as they're queried
+        # TODO: sanitize get parameter inputs?
+        try:
+            filter_val = parse_filter_conditions(request.GET.get("filter"))
+        except Exception:
+            return Response(detail=SCIM_400_INVALID_FILTER, status=400)
+
+        queryset = (
+            OrganizationMember.objects.filter(
+                Q(invite_status=InviteStatus.APPROVED.value),
+                Q(user__is_active=True) | Q(user__isnull=True),
+                organization=organization,
+            )
+            .select_related("user")
+            .order_by("email", "user__email")
+        )
+        if filter_val:
+            queryset = queryset.filter(
+                Q(email__in=filter_val) | Q(user__email__in=filter_val)
+            )  # not including secondary email vals (dups, etc.)
+
+        # TODO: how is queryset ordered?
```

Comment:
> how is queryset ordered?

My assumption would be that it's not well-defined or stable -- i.e., however Postgres feels like it.

Reply:
👍🏼 will remove this TODO -- SCIM protocol doesn't care about how its ordered, as long as its consistent, and I added an order by email to the above queryset.